src/lib.rs

```diff
@@ -73,9 +81,98 @@ fn map_spl_instructions(params: String, transactions: solTransactions) -> Result
 
         instructions.extend(output_instructions.instructions);
     }
+
+    let mut accounts_to_lookup = Vec::<Vec<u8>>::new();
+
+    for instruction in &instructions {
+        if let Some(ref item) = instruction.item {
+            match item {
+                Item::Transfer(transfer) => {
+                    if let Ok(from_bytes) = bs58::decode(&transfer.from).into_vec() {
+                        accounts_to_lookup.push(from_bytes);
+                    }
+                    if let Ok(to_bytes) = bs58::decode(&transfer.to).into_vec() {
+                        accounts_to_lookup.push(to_bytes);
+                    }
+                }
+                Item::Mint(mint) => {
+                    if let Ok(to_bytes) = bs58::decode(&mint.to).into_vec() {
+                        accounts_to_lookup.push(to_bytes);
+                    }
+                }
+                Item::Burn(burn) => {
+                    if let Ok(from_bytes) = bs58::decode(&burn.from).into_vec() {
+                        accounts_to_lookup.push(from_bytes);
+                    }
+                }
+                _ => {}
+            }
+        }
+    }
+
+    // Remove duplicates, better way ?
+    accounts_to_lookup.sort();
+    accounts_to_lookup.dedup();
+
+    let block_hash_bytes = bs58::decode(&block.blockhash).into_vec().unwrap_or_default();
+    let owners = get_account_owners_all(&foundational_store, &accounts_to_lookup, &block_hash_bytes, block.slot);
+
+    for instruction in &mut instructions {
+        if let Some(ref mut item) = instruction.item {
+            match item {
+                Item::Transfer(ref mut transfer) => {
+                    if let Some(from_owner) = owners.get(&transfer.from) {
+                        transfer.from_owner = from_owner.clone();
+                    }
+                    if let Some(to_owner) = owners.get(&transfer.to) {
+                        transfer.to_owner = to_owner.clone();
+                    }
+                }
+                Item::Mint(ref mut mint) => {
+                    if let Some(to_owner) = owners.get(&mint.to) {
+                        mint.to_owner = to_owner.clone();
+                    }
+                }
+                Item::Burn(ref mut burn) => {
+                    if let Some(from_owner) = owners.get(&burn.from) {
+                        burn.from_owner = from_owner.clone();
+                    }
+                }
+                _ => {}
+            }
+        }
+    }
+
     Ok(SplInstructions { instructions })
 }
 
+fn get_account_owners_all(
```

Comment:
```suggestion
fn get_account_owners(
```

Or `resolve_account_owners`, don't feel the `_all` is needed now that we are calling in a single batch.